### PR TITLE
fix(bug-report): URLs too long

### LINKED ISSUE #749: bug-report command creates too long request URI
#### Current Behavior
As i run `starship bug-report`, GitHub gets opened in the browser. Unfortunately the request URI seems to be too long for GitHub:

```
414 Request-URI Too Large - nginx
```

[Click here for the full request URI](https://github.com/starship/starship/issues/new?template=Bug_report.md&body=%3C%21--%0A%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20This%20issue%20has%20been%20pre-populated%20with%20your%20system%27s%20configuration%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%E2%99%A5%20Thank%20you%20for%20submitting%20a%20bug%20report%20%E2%99%A5%0A%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%E2%94%80%0A--%3E%0A%20%20%20%20%0A%23%23%20Bug%20Report%0A%0A%23%23%23%23%20Current%20Behavior%0A%3C%21--%20A%20clear%20and%20concise%20description%20of%20the%20behavior.%20--%3E%0A%0A%23%23%23%23%20Expected%20Behavior%0A%3C%21--%20A%20clear%20and%20concise%20description%20of%20what%20you%20expected%20to%20happen.%20--%3E%0A%0A%23%23%23%23%20Additional%20context%2FScreenshots%0A%3C%21--%20Add%20any%20other%20context%20about%20the%20problem%20here.%20If%20applicable%2C%20add%20screenshots%20to%20help%20explain.%20--%3E%0A%0A%23%23%23%23%20Possible%20Solution%0A%3C%21---%20Only%20if%20you%20have%20suggestions%20on%20a%20fix%20for%20the%20bug%20--%3E%0A%0A%23%23%23%23%20Environment%0A-%20Starship%20version%3A%200.30.1%0A-%20bash%20version%3A%20GNU%20bash%2C%20Version%204.4.20%281%29-release%20%28x86_64-pc-linux-gnu%29%0ACopyright%20%28C%29%202016%20Free%20Software%20Foundation%2C%20Inc.%0ALizenz%20GPLv3%2B%3A%20GNU%20GPL%20Version%203%20oder%20j%C3%BCnger%20%3Chttp%3A%2F%2Fgnu.org%2Flicenses%2Fgpl.html%3E%0A%0ADies%20ist%20freie%20Software.%20%20Sie%20darf%20ver%C3%A4ndert%20und%20verteilt%20werden.%0AEs%20wird%20keine%20Garantie%20gew%C3%A4hrt%2C%20soweit%20das%20Gesetz%20es%20zul%C3%A4sst.%0A-%20Operating%20system%3A%20Ubuntu%2018.04%0A%0A%23%23%23%23%20Relevant%20Shell%20Configuration%0A%0A%60%60%60bash%0A%23%20~%2F.bashrc%3A%20executed%20by%20bash%281%29%20for%20non-login%20shells.%0A%23%20see%20%2Fusr%2Fshare%2Fdoc%2Fbash%2Fexamples%2Fstartup-files%20%28in%20the%20package%20bash-doc%29%0A%23%20for%20examples%0A%0A%23%20If%20not%20running%20interactively%2C%20don%27t%20do%20anything%0Acase%20%24-%20in%0A%20%20%20%20%2Ai%2A%29%20%3B%3B%0A%20%20%20%20%20%20%2A%29%20return%3B%3B%0Aesac%0A%0A%23%20don%27t%20put%20duplicate%20lines%20or%20lines%20starting%20with%20space%20in%20the%20history.%0A%23%20See%20bash%281%29%20for%20more%20options%0AHISTCONTROL%3Dignoreboth%0A%0A%23%20append%20to%20the%20history%20file%2C%20don%27t%20overwrite%20it%0Ashopt%20-s%20histappend%0A%0A%23%20for%20setting%20history%20length%20see%20HISTSIZE%20and%20HISTFILESIZE%20in%20bash%281%29%0AHISTSIZE%3D1000%0AHISTFILESIZE%3D2000%0A%0A%23%20check%20the%20window%20size%20after%20each%20command%20and%2C%20if%20necessary%2C%0A%23%20update%20the%20values%20of%20LINES%20and%20COLUMNS.%0Ashopt%20-s%20checkwinsize%0A%0A%23%20If%20set%2C%20the%20pattern%20%22%2A%2A%22%20used%20in%20a%20pathname%20expansion%20context%20will%0A%23%20match%20all%20files%20and%20zero%20or%20more%20directories%20and%20subdirectories.%0A%23shopt%20-s%20globstar%0A%0A%23%20make%20less%20more%20friendly%20for%20non-text%20input%20files%2C%20see%20lesspipe%281%29%0A%5B%20-x%20%2Fusr%2Fbin%2Flesspipe%20%5D%20%26%26%20eval%20%22%24%28SHELL%3D%2Fbin%2Fsh%20lesspipe%29%22%0A%0A%23%20set%20variable%20identifying%20the%20chroot%20you%20work%20in%20%28used%20in%20the%20prompt%20below%29%0Aif%20%5B%20-z%20%22%24%7Bdebian_chroot%3A-%7D%22%20%5D%20%26%26%20%5B%20-r%20%2Fetc%2Fdebian_chroot%20%5D%3B%20then%0A%20%20%20%20debian_chroot%3D%24%28cat%20%2Fetc%2Fdebian_chroot%29%0Afi%0A%0A%23%20set%20a%20fancy%20prompt%20%28non-color%2C%20unless%20we%20know%20we%20%22want%22%20color%29%0Acase%20%22%24TERM%22%20in%0A%20%20%20%20xterm-color%7C%2A-256color%29%20color_prompt%3Dyes%3B%3B%0Aesac%0A%0A%23%20uncomment%20for%20a%20colored%20prompt%2C%20if%20the%20terminal%20has%20the%20capability%3B%20turned%0A%23%20off%20by%20default%20to%20not%20distract%20the%20user%3A%20the%20focus%20in%20a%20terminal%20window%0A%23%20should%20be%20on%20the%20output%20of%20commands%2C%20not%20on%20the%20prompt%0A%23force_color_prompt%3Dyes%0A%0Aif%20%5B%20-n%20%22%24force_color_prompt%22%20%5D%3B%20then%0A%20%20%20%20if%20%5B%20-x%20%2Fusr%2Fbin%2Ftput%20%5D%20%26%26%20tput%20setaf%201%20%3E%26%2Fdev%2Fnull%3B%20then%0A%09%23%20We%20have%20color%20support%3B%20assume%20it%27s%20compliant%20with%20Ecma-48%0A%09%23%20%28ISO%2FIEC-6429%29.%20%28Lack%20of%20such%20support%20is%20extremely%20rare%2C%20and%20such%0A%09%23%20a%20case%20would%20tend%20to%20support%20setf%20rather%20than%20setaf.%29%0A%09color_prompt%3Dyes%0A%20%20%20%20else%0A%09color_prompt%3D%0A%20%20%20%20fi%0Afi%0A%0Aif%20%5B%20%22%24color_prompt%22%20%3D%20yes%20%5D%3B%20then%0A%20%20%20%20PS1%3D%27%24%7Bdebian_chroot%3A%2B%28%24debian_chroot%29%7D%5C%5B%5C033%5B01%3B32m%5C%5D%5Cu%40%5Ch%5C%5B%5C033%5B00m%5C%5D%3A%5C%5B%5C033%5B01%3B34m%5C%5D%5Cw%5C%5B%5C033%5B00m%5C%5D%5C%24%20%27%0Aelse%0A%20%20%20%20PS1%3D%27%24%7Bdebian_chroot%3A%2B%28%24debian_chroot%29%7D%5Cu%40%5Ch%3A%5Cw%5C%24%20%27%0Afi%0Aunset%20color_prompt%20force_color_prompt%0A%0A%23%20If%20this%20is%20an%20xterm%20set%20the%20title%20to%20user%40host%3Adir%0Acase%20%22%24TERM%22%20in%0Axterm%2A%7Crxvt%2A%29%0A%20%20%20%20PS1%3D%22%5C%5B%5Ce%5D0%3B%24%7Bdebian_chroot%3A%2B%28%24debian_chroot%29%7D%5Cu%40%5Ch%3A%20%5Cw%5Ca%5C%5D%24PS1%22%0A%20%20%20%20%3B%3B%0A%2A%29%0A%20%20%20%20%3B%3B%0Aesac%0A%0A%23%20enable%20color%20support%20of%20ls%20and%20also%20add%20handy%20aliases%0Aif%20%5B%20-x%20%2Fusr%2Fbin%2Fdircolors%20%5D%3B%20then%0A%20%20%20%20test%20-r%20~%2F.dircolors%20%26%26%20eval%20%22%24%28dircolors%20-b%20~%2F.dircolors%29%22%20%7C%7C%20eval%20%22%24%28dircolors%20-b%29%22%0A%20%20%20%20alias%20ls%3D%27ls%20--color%3Dauto%27%0A%20%20%20%20%23alias%20dir%3D%27dir%20--color%3Dauto%27%0A%20%20%20%20%23alias%20vdir%3D%27vdir%20--color%3Dauto%27%0A%0A%20%20%20%20alias%20grep%3D%27grep%20--color%3Dauto%27%0A%20%20%20%20alias%20fgrep%3D%27fgrep%20--color%3Dauto%27%0A%20%20%20%20alias%20egrep%3D%27egrep%20--color%3Dauto%27%0Afi%0A%0A%23%20colored%20GCC%20warnings%20and%20errors%0A%23export%20GCC_COLORS%3D%27error%3D01%3B31%3Awarning%3D01%3B35%3Anote%3D01%3B36%3Acaret%3D01%3B32%3Alocus%3D01%3Aquote%3D01%27%0A%0A%23%20some%20more%20ls%20aliases%0Aalias%20ll%3D%27ls%20-alF%27%0Aalias%20la%3D%27ls%20-A%27%0Aalias%20l%3D%27ls%20-CF%27%0A%0Aalias%20ts%3D%27~%2FDownloads%2FTeamSpeak3-Client-linux_amd64%2Fts3client_runscript.sh%27%0Aalias%20gl%3D%27~%2FDownloads%2FGoLand-2019.2%2Fbin%2Fgoland.sh%27%0Aalias%20m13%3D%27java%20-jar%20%24HOME%2FDownloads%2FM13%2FWhereIsM13.jar%27%0A%0A%0A%23%20Add%20an%20%22alert%22%20alias%20for%20long%20running%20commands.%20%20Use%20like%20so%3A%0A%23%20%20%20sleep%2010%3B%20alert%0Aalias%20alert%3D%27notify-send%20--urgency%3Dlow%20-i%20%22%24%28%5B%20%24%3F%20%3D%200%20%5D%20%26%26%20echo%20terminal%20%7C%7C%20echo%20error%29%22%20%22%24%28history%7Ctail%20-n1%7Csed%20-e%20%27%5C%27%27s%2F%5E%5Cs%2A%5B0-9%5D%5C%2B%5Cs%2A%2F%2F%3Bs%2F%5B%3B%26%7C%5D%5Cs%2Aalert%24%2F%2F%27%5C%27%27%29%22%27%0A%0A%23%20Alias%20definitions.%0A%23%20You%20may%20want%20to%20put%20all%20your%20additions%20into%20a%20separate%20file%20like%0A%23%20~%2F.bash_aliases%2C%20instead%20of%20adding%20them%20here%20directly.%0A%23%20See%20%2Fusr%2Fshare%2Fdoc%2Fbash-doc%2Fexamples%20in%20the%20bash-doc%20package.%0A%0Aif%20%5B%20-f%20~%2F.bash_aliases%20%5D%3B%20then%0A%20%20%20%20.%20~%2F.bash_aliases%0Afi%0A%0A%23%20enable%20programmable%20completion%20features%20%28you%20don%27t%20need%20to%20enable%0A%23%20this%2C%20if%20it%27s%20already%20enabled%20in%20%2Fetc%2Fbash.bashrc%20and%20%2Fetc%2Fprofile%0A%23%20sources%20%2Fetc%2Fbash.bashrc%29.%0Aif%20%21%20shopt%20-oq%20posix%3B%20then%0A%20%20if%20%5B%20-f%20%2Fusr%2Fshare%2Fbash-completion%2Fbash_completion%20%5D%3B%20then%0A%20%20%20%20.%20%2Fusr%2Fshare%2Fbash-completion%2Fbash_completion%0A%20%20elif%20%5B%20-f%20%2Fetc%2Fbash_completion%20%5D%3B%20then%0A%20%20%20%20.%20%2Fetc%2Fbash_completion%0A%20%20fi%0Afi%0A%0Aeval%20%22%24%28starship%20init%20bash%29%22%0A%60%60%60%0A%0A%23%23%23%23%20Starship%20Configuration%0A%0A%60%60%60toml%0Aadd_newline%20%3D%20false%0A%0A%60%60%60)

#### Expected Behavior
The issue page should be displayed properly.

#### Additional context/Screenshots
image

#### Environment
Not relevant since the error comes from GitHub's nginx

### PR DESCRIPTION
**Pull request by jonstodle**
This fixes the issue reported in #749 

#### Description

By only using a simple template in the url and instead copy the environment info to the clipboard, I think we can land on a nice middle ground between not loosing to much of the convenience while still being able to handle large configs.

A quick search on the web tells me that most browsers are happy as long as URLs are not much longer than ~2000 character, which may present some problems in regards to larger configs. The URL linked in the issue is almost 10,000 characters long.

#### Motivation and Context

Closes #749

#### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### How Has This Been Tested?

This works consistently when testing with HUGE configs (300,000+ characters).
- [x] I have tested using **MacOS**
- [ ] I have tested using **Linux**
- [ ] I have tested using **Windows**

#### Checklist:

- [ ] I have updated the documentation accordingly.
- [x] I have updated the tests accordingly.

### CODE REVIEW

File: src/bug_report.rs [hunk 1]
```diff
@@ -30,10 +36,17 @@ pub fn create() {
             .unwrap_or(link);
 
         println!(
-            "Click this link to create a GitHub issue populated with your configuration:\n\n  {}",
+            "Your environment info has been copied to your clipboard. Click this link to create a GitHub issue:\n\n  {}",
             link
         );
     }
+
+    if copy_success {
```
Comment (Scotsguy) [inline src/bug_report.rs]: This reads as "if copying into the clipboard was successful, print 'Your clipboard was unavailable'". Shouldn't this be the other way around? (Also see `is_err`)

Comment (jonstodle) [reply src/bug_report.rs]: I just noticed this too...

File: docs/config/README.md [hunk 2]
```diff
@@ -61,17 +61,20 @@ This is the list of prompt-wide configuration options.
...
+| `add_newline`           | `true`                        | Add a new line before the start of the prompt.               |
+| `prompt_order`          | [link](#default-prompt-order) | Configure the order in which the prompt module occurs.       |
+| `scan_timeout`          | `30`                          | Timeout for starship to scan files (in milliseconds).        |
 
 ### Example
 
 ```toml
 # ~/.config/starship.toml
 
+# Clear the terminal from the prompt to the bottom of the terminal before printing.
```
Comment (matchai) [inline docs/config/README.md]: It looks like changes from #739 might've gotten mixed into this PR.

Comment (jonstodle) [reply docs/config/README.md]: I apparently managed to fork off of the wrong branch 🙄. Last night was apparently not a good coding moment for me.

**Comment by jonstodle** [comment]
I'm not sure what makes the Ubuntu tests fail. I can't really tell what the error is trying to tell me.

My only theory is that I've added a new dependency on [`clipboard`](https://crates.io/crates/clipboard), which mentions in it's doc that [xorg-dev](https://github.com/aweinstock314/rust-clipboard#prerequisites) is required for it to work.

**Comment by chipbuster** [comment]
@jonstodle It looks like your hunch is right:

```
  = note: /usr/bin/ld: cannot find -lxcb-shape
          /usr/bin/ld: cannot find -lxcb-xfixes
```

`libxcb` is the client-side implementation of the X11 protocol.

Is there a way to remove the dependency on `clipboard`? I currently run `starship` on at least three different computers that have no X11 libraries installed (two of which I cannot install them on), and it seems to be a bit of a stretch to require GUI libraries to run a shell prompt.

File: src/bug_report.rs [hunk 3]
```diff
@@ -106,13 +82,22 @@ fn format_env_info(starship_version: &str, environment: Environment) -> String {
...
         starship_config = environment.starship_config,
+    ))
+        .replace("%20", "+");
+
+    format!(
+        "https://github.com/starship/starship/issues/new?body={}",
+        body
     )
+    .chars()
+    .take(8100) // Magic number accepted by Github
```
Comment (Scotsguy) [inline src/bug_report.rs]: Make 8100 a `const`, although I'm not sure just cutting off everything past 8100 chars is the right way to do this

Comment (jonstodle) [reply src/bug_report.rs]: That makes two of us, but I figured it's better than making Github throw an error.

File: src/bug_report.rs [hunk 4]
```diff
@@ -99,12 +89,16 @@ fn make_github_issue_link(starship_version: &str, environment: Environment) -> S
         os_version = environment.os_version,
         shell_config = environment.shell_info.config,
         starship_config = environment.starship_config,
-    ));
+    ))
+        .replace("%20", "+");
 
     format!(
-        "https://github.com/starship/starship/issues/new?template={}&body={}",
```
Comment (matchai) [inline src/bug_report.rs]: I believe the template query param here is required for the bug label to apply to the new issue.

**Comment by matchai** [comment]
Thank you for fixing this up, @jonstodle! 😊

**Pull request merged**